Add disqualified state for the form-wizard. (#1274)

mikeconley opened the pull request.

hannajones commented on kitsune/sumo/static/sumo/js/form-wizard.js:
```diff
@@ -150,8 +160,15 @@ export class FormWizard extends HTMLElement {
   }
 
   // To be implemented as part of mozilla/sumo #1274.
```
```suggestion
```

Done!

hannajones commented on kitsune/sumo/static/sumo/js/form-wizard.js:
```diff
@@ -34,14 +35,23 @@ export class FormWizard extends HTMLElement {
+          <progress max="100" value="0"></progress>
+          <div class="form-wizard-disqualified">
+            <div class="disqualification">
+              <img class="warning-icon" src="${warningImageURL}"/>
```
```suggestion
              <img class="warning-icon" src="${warningImageURL}" aria-hidden="true" />
```

Let's keep this icon thing going

hannajones commented on kitsune/sumo/static/sumo/js/form-wizard.js:
```diff
@@ -34,14 +35,23 @@ export class FormWizard extends HTMLElement {
+          <progress max="100" value="0"></progress>
+          <div class="form-wizard-disqualified">
+            <div class="disqualification">
+              <img class="warning-icon" src="${warningImageURL}"/>
+              <span class="disqualification-header"></span>
```
<img width="215" alt="Screenshot 2023-05-18 at 5 41 11 PM" src="https://github.com/mozilla/kitsune/assets/15138046/11d961e7-f28d-45b8-8e82-05c0f1f508cb">


Splitting hairs here a bit, but the icon isn't quite vertically centered with the text. Maybe worth wrapping them in a flex container?

hannajones reviewed (approved):
Thanks! Couple of small suggestions, but this looks great! I even tested it in Chrome and we're dong what we need to do there 😎

mikeconley commented:
Hey @hannajones! Thanks for the review. I think I pushed a slight alteration just as you were publishing it! I've changed things slightly, where now the strings for the disqualification messages are defined in the markup.

I also added the UITour-is-broken disqualification message even though it isn't being used just yet - mainly to optimize for getting our strings into the tree. I'll wire it up to UITour failure tomorrow.

hannajones reviewed (approved):
LGTM, thanks!

flodolo commented on kitsune/sumo/static/sumo/js/form-wizard.js:
```diff
@@ -34,18 +35,31 @@ export class FormWizard extends HTMLElement {
+            </div>
+
+            <div class="disqualification" reason="uitour-broken">
+              <div class="disqualification-message">${gettext("The setup assistant is currently unavailable for your version of Firefox. However, you can still perform a manual backup of your data by following the steps outlined in")} <a href="/kb/profile-manager-create-remote-switch-firefox-profiles">${gettext("this article.")}</a></div>
```
Please never do this.
https://mozilla-l10n.github.io/documentation/localization/dev_best_practices.html#avoid-concatenations-use-placeholders-instead

These two strings landed as separate strings with zero context, I assume concatenation only because `this article.` stars with lowercase and ends with a period.

This is effectively impossible to localize if a language requires "this article" in the middle of the other sentence.

mikeconley replied:
Thanks, @flodolo. What is the correct way to localize this in the JavaScript layer of this Django application?

escattone replied:
@flodolo 👍 That makes sense. I'm at fault for recommending this approach.

hannajones replied:
hey @flodolo let me know if this approach using `interpolate` helps address the issue: https://github.com/mozilla/kitsune/pull/5517

if not I'm open to other suggestions!
